identity-iota/src/tangle/publish.rs

```diff
@@ -200,21 +198,25 @@ mod test {
   fn test_publish_type_update_method_with_non_update_method_type2() -> Result<()> {
     let mut old_doc = document();
 
-    let collection = KeyCollection::new_ed25519(8)?;
+    let keypair: KeyPair = KeyPair::new(KeyType::X25519)?;
     let method: IotaVerificationMethod =
-      IotaVerificationMethod::new_merkle_key::<Sha256>(old_doc.id().to_owned(), &collection, "merkle")?;
+      IotaVerificationMethod::new(old_doc.id().to_owned(), keypair.type_(), keypair.public(), "kex-0")?;
 
     old_doc
       .insert_method(method, MethodScope::capability_invocation())
       .unwrap();
 
     let mut new_doc = old_doc.clone();
 
-    // Replace the key collection.
-    let new_collection = KeyCollection::new_ed25519(8)?;
+    // Replace the key material in the new method.
+    let keypair_new: KeyPair = KeyPair::new(KeyType::X25519)?;
 
-    let method_new: IotaVerificationMethod =
-      IotaVerificationMethod::new_merkle_key::<Sha256>(new_doc.id().to_owned(), &new_collection, "merkle")?;
+    let method_new: IotaVerificationMethod = IotaVerificationMethod::new(
+      new_doc.id().to_owned(),
+      keypair_new.type_(),
+      keypair_new.public(),
+      "kex-0",
```

Comment:
typo?

Reply:
Are you referring to the name `"kex-0"`? If so, no it's intentional, `"kex"` = key-exchange.